apps/service_providers/llm_service/runnables.py

```diff
@@ -304,13 +304,21 @@ def _sync_messages_to_thread(self, current_thread_id):
 
         This is necessary in multi-bot setups if some of the bots are assistants but not all.
         """
+
+        def __sync_messages_to_thread(thread_id, messages):
+            for message in messages_to_sync:
+                self.state.raw_client.beta.threads.messages.create(current_thread_id, **message)
+
         if messages_to_sync := self.state.get_messages_to_sync_to_thread():
             if current_thread_id:
-                for message in messages_to_sync:
-                    self.state.raw_client.beta.threads.messages.create(current_thread_id, **message)
+                __sync_messages_to_thread(current_thread_id, messages_to_sync)
             else:
-                thread = self.state.raw_client.beta.threads.create(messages=messages_to_sync)
+                # There is a 32 message limit when creating a new thread
+                first, rest = messages_to_sync[:32], messages_to_sync[32:]
+                thread = self.state.raw_client.beta.threads.create(messages=first)
                 current_thread_id = thread.id
+                if rest:
```

Comment:
Super nitty comment: You don't have to check if there is a `rest` since it will not iterate over anything in the `__sync_messages_to_thread` method.

Question: Why the double underscore and not a single one?

Reply:
> Question: Why the double underscore and not a single one?

The outer method is called the same thing (with a single underscore) and I was too lazy to think of another name.